crates/ruff_python_ast/src/visitor.rs

```diff
@@ -633,10 +633,10 @@ pub fn walk_arg_with_default<'a, V: Visitor<'a> + ?Sized>(
     visitor: &mut V,
     arg_with_default: &'a ArgWithDefault,
 ) {
-    visitor.visit_arg(&arg_with_default.def);
     if let Some(expr) = &arg_with_default.default {
         visitor.visit_expr(expr);
     }
+    visitor.visit_arg(&arg_with_default.def);
```

Comment:
Yeah, that's the way I originally implemented it, but after resolving the conflicts from #5192, this seemed to be the simplest way to do things. I'm open to restructuring, but it would make this PR more involved.

Reply:
Yes, you're right. This doesn't completely solve the issue. But, given as given as function arguments are now walked one by one, it wouldn't be possible to traverse them in the correct order without removing or avoiding calling `visit_arg_with_default`. I think the simplest solution would be to have `walk_arguments` handle all argument traversal and never call `visit_arg_with_default`, but this isn't the cleanest in my opinion. I'll through this change in though and you can let me know what you think.

Reply:
Hmm, this became unbearably messy for my tastes. The problem is we need to call `visit_arg` with a `ArgWithDefault`, so we have to convert an `ArgWithDefault` to just a plain `Arg`... but then there are lifetime issues to resolve and the whole thing blows up. I'm inclined to leave this as is for now. Yes, it's not completely correct, but it's so subtle that I don't anticipate it causing issues.

Reply:
Hah, ok -- that's fine, lets see how it goes, this is clearly an improvement.